### PR TITLE
CO-3145 first half of l10n fix direct debit xml generation

### PR DESCRIPTION
**Pull request by quentingigon**

### CODE REVIEW

**Comment by quentingigon** [comment]
Second half is [there](https://github.com/CompassionCH/bank-payment/pull/2)

File: l10n_ch_pain_direct_debit/__manifest__.py [hunk 1]
```diff
@@ -29,3 +29,5 @@
     'auto_install': False,
     'installable': True,
 }
+
+# old postal account for res_partner_bank(4032): 010306797
```
Comment (ecino) [inline l10n_ch_pain_direct_debit/__manifest__.py]: Can you remove this?

File: l10n_ch_pain_direct_debit/models/account_payment_order.py [hunk 2]
```diff
@@ -413,12 +418,13 @@ def generate_xml_ch_dd_file(self):
                     "self.name + '-' "
                     "+ str(requested_date) + '-' + priority + "
                     "'-' + local_instrument",
-                    prio, local_inst, False, False, req_date, {
+                    # TODO replace the False just before eval_ctx by sepa boolean
```
Comment (ecino) [inline l10n_ch_pain_direct_debit/models/account_payment_order.py]: What's this? We don't want to leave todos in OCA repos.

**Review by ecino — CHANGES_REQUESTED** [review]
Please fix flake8 issues in module:
```
l10n_ch_pain_direct_debit/models/account_payment_order.py:120:9: E125 continuation line with same indent as next logical line
l10n_ch_pain_direct_debit/models/account_payment_order.py:450:89: E501 line too long (94 > 88 characters)
```

File: l10n_ch_pain_direct_debit/models/account_payment_order.py (hunk 2 above)
Comment (quentingigon) [reply l10n_ch_pain_direct_debit/models/account_payment_order.py]: Ah yes I forgot to remove that

**Comment by quentingigon** [comment]
Some tests of some modules in `l10n_switzerland` repo are failing but not the current module

**Comment by quentingigon** [comment]
This is the code making the error:
`action = self.env['ir.actions.act_window'].for_xml_id(
            'account_payment_order',
            'account_payment_order_%s_action' % action_payment_type)`
line 126 in `bank-payement/account_payement_order/account_invoice.py`

It asks for an xml_id : `account_payment_order_debit_action` that does not exists and I have no idea what to replace it with. Any advice ?

**Review by ecino — APPROVED** [review]

**Comment by ecino** [comment]
Manually merged into branch. Thanks.

**Pull request closed without merging**